Source/WebCore/platform/text/WritingMode.h

```diff
@@ -80,6 +80,11 @@ constexpr unsigned makeTextFlowInitalizer(BlockFlowDirection blockFlow, TextDire
     return static_cast<unsigned>(blockFlow) << 1 | static_cast<unsigned>(direction);
 }
 
+struct WritingFlow {
```

Comment:
This is very similar to TextFlow. We should probably re-adapt that instead? You'd want to check against the BlockFlowDirection rather than the WritingMode too.

Reply:
I agree, I've actually was using `TextFlow` in the beginning, but it was making me a bit confused, I always was trying to mentally map TextFlow -> BlockDirection -> WritingMode. So think this is a better abstraction to maintain, IMO. It is easier to understand what is going on when accessing WritingMode and direction explicitly.

Reply:
@alanbaradlay and I discussed this and we think we should try to refactor TextFlow to be less confusing instead of introducing something new that's almost identical. There's very few usages of makeTextFlow so it shouldn't be too hard: https://searchfox.org/wubkat/search?q=makeTextFlow&path=&regexp=false

Reply:
There are more references to `TextFlow`, so I'll just use `TextFlow` as it is here.